google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/DefaultCloudSdkService.java

```diff
@@ -35,6 +35,11 @@
   private PropertiesComponent propertiesComponent = PropertiesComponent.getInstance();
   private static final String CLOUD_SDK_PROPERTY_KEY = "GCT_CLOUD_SDK_HOME_PATH";
 
+  @Override
+  public void activate() {
```

Comment:
is this just to be used for metrics?

Reply:
For `ManagedSDKService`, it's a notification to check if installation is needed (i.e. first time install, broken existing managed SDK etc.). For default service, nothing except metrics so far.

Reply:
do you mind elaborating a bit? how is this different than `install`; what is the workflow that calls `active` / `install`; etc. `install` is intuitive to me, but `activate` isn't in this context

Reply:
We are going to have two implementations - custom and managed SDK services. When a user goes to settings and changes implementation, `activate()` is called for newly selected and applied service. `install()` is called by it only by `ManagedSDKService`. `install()` will also be called by deployment / run configuration explicitly if status is not `READY` in attempt to prepare SDK immediately without going to settings.

Reply:
oh i see! i forgot that the user can switch between managed and custom